contents/all.tsx

```diff
@@ -7,19 +7,40 @@ export const config: PlasmoContentScript = {
   matches: ["https://*/*"]
 }
 
+const DIV_ID = "plasmo-shadow-div"
 const SHADOW_STYLE_ID = "shadow-styling"
+
+// We use the getRootContainer to inject the shadow host in our own way
+export const getRootContainer = async () => {
+  const container = document.createElement("div")
+
+  container.id = "plasmo-shadow-container"
+
+  container.style.cssText = `
+    display: none;
+    z-index: 1;
+    position: absolute;
+  `
+
+  // This is the div that the shadow host will be inserted into
+  const shadowHost = document.createElement("div")
+  shadowHost.id = DIV_ID
+
+  const shadowRoot = shadowHost.attachShadow({ mode: "open" })
+  document.body.insertAdjacentElement("beforebegin", shadowHost)
+
+  shadowRoot.appendChild(await getStyle())
+
+  shadowRoot.appendChild(container)
+  return container
+}
+
 export const getStyle = () => {
```

Comment:
something to keep in mind - when we use getRootContainer, getStyle will be ignored, but we can still use it (see line 32)

Reply:
> when we use getRootContainer, getStyle will be ignored

TIL!